R/fda-table_05.R

```diff
@@ -0,0 +1,81 @@
+# Pre-processing
+
+library(dplyr)
+library(scda)
+library(tern)
+library(lubridate)
+
+adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl") %>%
+  filter(SAFFL == "Y")
+
+adex <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adex") %>%
+  filter(SAFFL == "Y") %>%
+  select("USUBJID", "ARM", "TRTSDTM", "TRTEDTM") %>%
+  distinct() %>%
+  mutate(TRTS_D = as.Date(ymd_hms(TRTSDTM))) %>%
+  mutate(TRTE_D = as.Date(ymd_hms(TRTEDTM))) %>%
+  mutate(TRTDUR = as.numeric(TRTE_D - TRTS_D + 1)) %>%
+  mutate(
+    any_d = (TRTDUR > 0),
+    d_less_1 = (TRTDUR < 365 / 12 * 1),
+    d_more_1 = (TRTDUR >= 365 / 12 * 1),
+    d_more_3 = (TRTDUR >= 365 / 12 * 3),
+    d_more_6 = (TRTDUR >= 365 / 12 * 6),
+    d_more_12 = (TRTDUR >= 365 / 12 * 12)
+  ) %>%
+  formatters::var_relabel(
+    any_d = "Any duration (at least 1 dose)",
+    d_less_1 = "<1 month",
+    d_more_1 = ">=1 month",
+    d_more_3 = ">=3 months",
+    d_more_6 = ">=6 months",
+    d_more_12 = ">=12 months"
+  )
+
+# Build layout
+
+lyt <- basic_table(show_colcounts = TRUE) %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "TRTDUR", var_labels = "Duration of treatment, weeks (or months, or days, or cycles)", function(x, ...) {
```

Comment:
We can specify the unit since `TRTDUR` is in days for this dataset
```suggestion
  analyze(vars = "TRTDUR", var_labels = "Duration of treatment, days", function(x, ...) {
```